### PR TITLE
move Insights and Campaigns from global navbar to user menu

### PR DESCRIPTION
**Pull request by sqs**
- These links take up precious width in the global navbar. Often a search query will be long (for example, because it contains a repo name). In that case, it's much more valuable to have a wide query input than to show these links.
- The user menu previously contained some other similar links, such as "Extensions" and "Saved searches". It seems inconsistent to show those in the user menu but to show these others in the global navbar. Now they are all in the user menu, which means the user knows there's 1 place they can go for anything like this.

## Before

image

## After

image

### CODE REVIEW

**Review by mrnugget — COMMENTED** [review]
I'm +10 in favour of this.

(Only commenting, not approving, because I think the decision should be made by design/product/web)

**Review by tjkandala — APPROVED** [review]
LGTM (for now). cc @AlicjaSuska this will change with #13372, right?

**Comment by AlicjaSuska** [comment]
Discussion about this is taking place in the [Slack thread](https://sourcegraph.slack.com/archives/CHEKCRWKV/p1600180166095700)

Summary:
- the design team is against removing the tabs because of discoverability issues
- we recognize a need to accommodate longer searches
- adding a vertical navbar seems to be a reasonable option that solves both problems

**Comment by rrhyne** [comment]
A core user experience principle is that Icons must be accompanied by text. Great article describing why here: https://www.nngroup.com/articles/icon-usability/

TLDR; Each user can bring a different connotation to the icon. 

Important for us in our main navigation as each item conveys the most important aspects of Sourcegraph. 

That said, a collapsed menu is fine after the user learns the meaning of our icons.

**Comment by sqs** [comment]
> the design team is against removing the tabs because of discoverability issues

In the meantime before we have new nav, can you help me understand this more?

It seems like we have chosen some features (campaigns, code insights, code monitoring, explore) to put in global nav, and some to put in the user menu (saved searches, settings, organizations, site admin). This seems arbitrary.

Also, for discoverability, it's not clear that putting it in the top nav increases discoverability of things. (It certainly comes at a downside: clutter and narrower search input.) I know tons of people who miss the top nav links on GitHub, but they do find things that are added to the repo page or user profile page: 

image

**Comment by rrhyne** [comment]
Top-level nav links are more discoverable because they are visible on all pages and clearly labeled. Anything in a dropdown is hidden and must be actively discovered. Placing items in a dropdown is pretty universally used to house second or third team navigation items such as settings. Sites can and do break those 'rules' of course.

Top-level menu items should be the properties the organization feels are most important, and what is used most often. This is the document the design and web teams worked on to define that list:

https://docs.google.com/document/d/1bU-tq1yM4cechfAnHXEibkebWzACeFVfaFFXoqGk0mA/edit#bookmark=id.ptvd59f0fsqv

**Comment by rrhyne** [comment]
I realized I did not provide why I believe this statement to be true: "Top-level menu items should be the properties the organization feels are most important". It's mostly about discoverability. 

New users will scan the app and attempt to quickly determine what they can accomplish with it. Hidden elements are not seen by many users. This causes a certain percentage of users to churn and never come back.  

Additionally, some experienced users only use the app for search. We want to expand their usage to other segments. Placing those items front and center helps us get users to click on them which helps us educate as to the functionality.  

What are the most important portions of the application to the business?

**Comment by AlicjaSuska** [comment]
I agree with Rob's points. When it comes to your questions about GitHub @sqs - they most probably miss the links on the dark background as they are outside of the main focus area - the white part with repository on. Also, GitHub has a lot going on in all 3 levels of top bars. I wouldn't say it means that top bar navigation doesn't work, I would rather treat GitHub as an exception as it's so overloaded.

When it comes to the vertical nav. Here is how it could look like an expanded (on hover) and collapsed - version for power users with the blue indicator only, accessible after enabling 'keep collapsed' toogle.

nav-expended
nav-min

**Review by eseliger — APPROVED** [review]
Campaigns change LGTM

**Comment by sqs** [comment]
Closing in favor of https://www.figma.com/file/1nF5duAbwo6P8OTike9b2g/New-menu-proposition-%5BReview%5D?node-id=260%3A11 from @AlicjaSuska
